File: clients/python/examples/tables-insert-many-vector.py
from astrapy import DataAPIClient
from astrapy.data_types import (
    DataAPIVector,
)

# Get an existing table
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
table = database.get_table("**TABLE_NAME**")

# Insert rows into the table
result = table.insert_many(
    [
        {
            "title": "Computed Wilderness",
            "author": "Ryan Eau",
            "summary_genres_vector": DataAPIVector([0.08, -0.62, 0.39]),
        },
        {
            "title": "Desert Peace",
            "author": "Walter Dray",
            "summary_genres_vector": DataAPIVector([0.12, 0.53, 0.32]),
        },
    ]
)

File: clients/csharp/examples/tables-insert-many-vector.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Tables;

namespace Examples;

public class Book
{
  [ColumnPrimaryKey(1)]
  [ColumnName("title")]
  public string Title { get; set; } = null!;

  [ColumnPrimaryKey(2)]
  [ColumnName("author")]
  public string Author { get; set; } = null!;

  [ColumnVector(3)]
  [ColumnName("summary_genres_vector")]
  public double[]? SummaryGenresVector { get; set; }
}

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var table = database.GetTable<Book>("**TABLE_NAME**");

    // Insert rows into the table
    var rows = new List<Book>()
    {
      new Book()
      {
        Title = "Computed Wilderness",
        Author = "Ryan Eau",
        SummaryGenresVector = new double[] { 0.08f, -0.62f, 0.39f },
      },
      new Book()
      {
        Title = "Desert Peace",
        Author = "Walter Dray",
        SummaryGenresVector = new double[] { 0.12f, 0.53f, 0.32f },
      },
    };

    await table.InsertManyAsync(rows);
  }
}
